LoRaEngine/LoraKeysManagerFacade/IoTCentralImp/IoTCentralDevice.cs

```diff
@@ -0,0 +1,26 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace LoraKeysManagerFacade.IoTCentralImp
+{
+    using LoraKeysManagerFacade.IoTCentralImp.Definitions;
+
+    internal class IoTCentralDevice : IDevice
+    {
+        private readonly Device deviceObject;
+        private readonly SymmetricKeyAttestation attestationObject;
+
+        public IoTCentralDevice(Device deviceObject, SymmetricKeyAttestation attestationObject, string assignedIoTHubHostname)
+        {
+            this.deviceObject = deviceObject;
+            this.attestationObject = attestationObject;
+            this.AssignedIoTHub = assignedIoTHubHostname;
+        }
+
+        public string PrimaryKey => this.attestationObject.SymmetricKey.PrimaryKey;
+
+        public string DeviceId => this.deviceObject.Id;
+
+        public string AssignedIoTHub { get; }
+    }
```

Comment:
Avoid keeping whole objects alive in fields if we only use/need some of their properties:

```suggestion
    internal sealed class IoTCentralDevice : IDevice
    {
        public IoTCentralDevice(Device deviceObject, SymmetricKeyAttestation attestationObject, string assignedIoTHubHostname)
        {
            DeviceId = deviceObject.Id;
            PrimaryKey = attestationObject.SymmetricKey.PrimaryKey;
            AssignedIoTHub = assignedIoTHubHostname;
        }

        public string PrimaryKey { get; }
        public string DeviceId { get; }
        public string AssignedIoTHub { get; }
    }
```

I would even go as far as sending just the needed values in the constructor.